### PR TITLE
Dockerfile for OrbitDB base image

### PR DESCRIPTION
**Pull request by vvp**
This PR defines the base image for running OrbitDB code in containers. It's something that I've personally been using for a while, so it can already be used locally for running tests or examples in isolated container. Besides the initial Dockerfile for building OrbitDB base images, there's some instructions how to use it with Docker. 

There is optimization left to do in base image itself to make it leaner (smaller nodejs-base image, multi-stage builds etc). Also I've only been using it on Linux where there are usually no problems with containers, so it'd be great to hear experiences from Docker for Mac/Windows users.

### CODE REVIEW

**Comment by haadcode** [comment]
Looks good to me but I'm not very familiar with Docker.

**Comment by thiagodelgado111** [comment]
I have (almost) no experience with Docker. If anything, I've used it to provide things I needed to work, like: database, email server, ipfs, etc. On the Dockerfile, the only thing that occurred to me is that maybe `package-lock.json` should be copied too? I was gonna mention using `npm ci` but I think it probably isn't relevant for the case

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+After building local image, run node.js-examples inside container:
+
+```bash
+docker run -ti --rm orbit-db npm run examples:node
+```
+
+## Why would you want to run OrbitDB in container?
+
+Containers are nice because as software execution environments they are:
+- Reproducible, which helps testing and development because you can revert container to original state by destroying it and creating it again,
```
Comment (RichardLitt) [inline docker/README.md]: revert the container

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- Isolated, which guarantees that external factors like npm versions, operating system version, or other installed software like native compilers do not affect the execution.
```
Comment (RichardLitt) [inline docker/README.md]: npm versions, the operating system version, or..

**Comment by vvp** [comment]
@RichardLitt Thanks for proof-reading, fixed! :+1: 

@thiagodelgado111 good thing in not copying that lock-file into image is that it lets you build multiple local Docker images from the same OrbitDB package.json but with different dependencies. With them it's easy to setup a continuous testing for catching breaking dependencies as soon as possible (given that npm install or npm test catches them :slightly_smiling_face: )

**Comment by haadcode** [comment]
Thanks for PR and the docker support @vvp! 👍❤️

**Pull request merged**